test/runnable/dbitfieldsdm.d

```diff
@@ -0,0 +1,196 @@
+S9 = x30200
+S14 = x300000201
+S15 = x201
+S18 = 1 should be 4
+A0 = x1
+---
+ */
+
+import core.stdc.stdio;
+
+int is64bit() { return size_t.sizeof == 8; }  // otherwise assume 32 bit
+
+/*************************************************************/
+
+struct T0  { ubyte x:1; };                       //
+struct T1  { short x:1; };                       //
+struct T2  { int x:1; };                         //
+struct T3  { ubyte a,b,c,d; long x:1; };         //
+struct T4  { ubyte a,b,c,d,e,f,g,h; long x:1; }; //
+struct T5  { ubyte a,b,c,d,e,f,g; long x:1; };   //
+struct S1  { long f:1; };                        //
+struct S2  { int x:1; int y:1; };                //
+struct S3  { short c; int x:1; uint y:1; };      //
+struct S4  { int x:1; short y:1; };              //
+struct S5  { short x:1; int y:1; };              //
+struct S6  { short x:1; short y:1; };            //
+struct S7  { short x:1; int y:1; long z:1; };    //
+struct S8  { ubyte a; ubyte b:1; short c:2; };   //
+struct S8A { ubyte b:1; short c:2; };            //
+struct S8B { ubyte a; short b:1; ubyte c:2; };   //
+struct S8C { ubyte a; int b:1; };                //
+struct S9  { ubyte a; ubyte b:2; short c:9; };   //
+struct S10 { };                                  // sizeof differs from C treatment
+struct S11 { int :0; };                          // sizeof differs from C treatment
+struct S12 { int :0; int x; };                   //
+struct S13 { uint x:12; uint x1:1; uint x2:1; uint x3:1; uint x4:1; int w; }; //
+struct S14 { ubyte a; ubyte b:4; int c:30; };    //
+struct S15 { ubyte a; ubyte b:2; int c:9; };     //
+struct S16 { int :32; };                         // sizeof differs from C treatment
+struct S17 { int a:32; };                        //
+struct S18 { ubyte a; long :0; ubyte b; };       //
+struct A0  { int a; long b:34, c:4; };           //
+struct A1  { int a; uint b:11; int c; };         //
+struct A2  { int a; uint b:11, c:5, d:16;        //
+             int e; };
+struct A3  { int a; uint b:11, c:5, :0, d:16;    //
+             int e; };
+struct A4  { int a:8; short b:7;                 //
+             uint c:29; };
+struct A5  { ubyte a:7, b:2; };                  //
+struct A6  { ubyte a:7; short b:2; };            //
+struct A7  { short a:8; int b:16; int c;         //
+             ubyte d:7; };
+struct A8  { short a:8; int b:16; int :0;        //
+             ubyte c:7; };
+struct A9  { ushort a:8; int b:16;               //
+             uint c:29; long d:9;
+             uint e:2, f:31; };
+struct A10 { ushort a:8; ubyte b; };             //
+struct A11 { ubyte a; int b:5, c:11, :0, d:8;    //
+             struct { int ee:8; } };
+
+int main()
+{
+    /* MS produces identical results for 32 and 64 bit compiles,
+     * DM is 32 bit only
+     */
+    printf("                DM |   MS |  P32 |  P64\n");
+    printf("T0  = %2d %d ||  1 1 |  1 1 |  1 1 |  1 1\n", cast(int)T0.sizeof, cast(int)T0.alignof);
```

Comment:
Wouldn't it be better to just use asserts here?

Reply:
It's just easier to do it this way. The text representations are easier to edit, compare and understand.

Reply:
It doesn't really help my end because the testsuite framework I use doesn't support comparing ad-hoc text, that can be worked around.